launcher/icons/IconList.cpp

```diff
@@ -77,10 +77,90 @@ IconList::IconList(const QStringList& builtinPaths, QString path, QObject* paren
 void IconList::sortIconList()
 {
     qDebug() << "Sorting icon list...";
-    std::sort(icons.begin(), icons.end(), [](const MMCIcon& a, const MMCIcon& b) { return a.m_key.localeAwareCompare(b.m_key) < 0; });
+    std::sort(icons.begin(), icons.end(), [](const MMCIcon& a, const MMCIcon& b) {
+        bool aIsSubdir = a.m_key.contains(QDir::separator());
+        bool bIsSubdir = b.m_key.contains(QDir::separator());
+        if (aIsSubdir != bIsSubdir) {
+            return !aIsSubdir;  // root-level icons come first
+        }
+        return a.m_key.localeAwareCompare(b.m_key) < 0;
+    });
     reindex();
 }
 
+// Helper function to add directories recursively
+bool IconList::addPathRecursively(const QString& path)
+{
+    QDir dir(path);
+    if (!dir.exists())
+        return false;
+
+    bool watching = false;
+
+    // Add the directory itself
+    watching = m_watcher->addPath(path);
+
+    // Add all subdirectories
+    QFileInfoList entries = dir.entryInfoList(QDir::Dirs | QDir::NoDotAndDotDot);
+    for (const QFileInfo& entry : entries) {
+        if (addPathRecursively(entry.absoluteFilePath())) {
+            watching = true;
+        }
+    }
+    return watching;
+}
+
+void IconList::removePathRecursively(const QString& path)
+{
+    QFileInfo file_info(path);
+    if (file_info.isFile()) {
+        // Remove the icon belonging to the file
+        QString key = m_dir.relativeFilePath(file_info.absoluteFilePath());
+        int idx = getIconIndex(key);
+        if (idx == -1)
+            return;
+
```

Comment:
This code only checks the existence of the icon, it doesn't remove it.
So either this is not intended(you need to add the remove code) or this code is not needed.